### PR TITLE
 UI - Add Upgrade Button to Resources

### LINKED ISSUE #3118: Allow users to upgrade resources through the UI
## Description

As a TRE Administrator/TRE Workspace Owner/Researcher 
I want to be able to upgrade resources that have been deployed through the UI,
so that I can easily keep them up-to-date and take advantage of new features and bug fixes. 

## Acceptance criteria
- [ ] List the versions that we can upgrade to when we GET a resource
- [ ] Allow users to choose the version they want to upgrade to within the resource card, based on their role.

### PR DESCRIPTION
**Pull request by yuvalyaron**
# Resolves #3118

## What is being addressed

Added `Upgrade` button to resources that have available template upgrades (only non major upgrades are considered)

## How is this addressed

- An Upgrade button was added for resources that have an `availableUpgrade` with `forceUpdateRequired=false` (non major upgrade)

image

gif

### CODE REVIEW

File: ui/app/src/components/shared/ConfirmUpgradeResource.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    <Dialog
+      hidden={false}
+      onDismiss={() => props.onDismiss()}
+      dialogContentProps={upgradeProps}
+      modalProps={modalProps}
+    >
+      {
+        loading === LoadingState.Ok &&
+        <>
+          <MessageBar messageBarType={MessageBarType.warning} >Upgrading the template version is irreversible</MessageBar>
```
Comment (jjgriff93) [inline ui/app/src/components/shared/ConfirmUpgradeResource.tsx:89]:
```suggestion
          <MessageBar messageBarType={MessageBarType.warning} >Upgrading the template version is irreversible.</MessageBar>
```

File: ui/app/src/components/shared/ConfirmUpgradeResource.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    titleAriaId: 'labelId',
+    subtitleAriaId: 'subTextId',
+    isBlocking: true,
+    styles: dialogStyles
+  };
+
+  const wsAuth = (props.resource.resourceType === ResourceType.WorkspaceService || props.resource.resourceType === ResourceType.UserResource);
+
+  const upgradeCall = async () => {
+    setLoading(LoadingState.Loading);
```
Comment (jjgriff93) [inline ui/app/src/components/shared/ConfirmUpgradeResource.tsx:44]: `nit` we tend to use "loading" and "loadingState" for indicating whether a page/component has loaded after navigation. For API request statuses, maybe call is `requestStatus` or something more specific to a request

**Review by jjgriff93 — APPROVED** [review]
Really great! UI for this looks very slick - just added some minor suggestions

File: ui/app/src/components/shared/ConfirmUpgradeResource.tsx (hunk 2 above)
Comment (yuvalyaron) [reply ui/app/src/components/shared/ConfirmUpgradeResource.tsx:44]: 👍  renamed to `requestLoadingState`

**Pull request merged**